test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/EnumFilterFunctionalTests.cs

```diff
@@ -515,28 +520,245 @@ public void ParseFilterEnumTypesWrongCast3()
+            bool success = colorType.TryParse(enumValue, out IEdmEnumMember enumMember);
+
+            // Act
+            FilterClause filterQueryNode = ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+            SingleValueNode expression = filterQueryNode.Expression;
+
+            // Assert
+            Assert.Equal("3 in Colors", filterQuery);
+            Assert.True(success);
+            InNode inNode = expression.ShouldBeInNode();
+            ConstantNode leftNode = Assert.IsType<ConstantNode>(inNode.Left);
+            leftNode.ShouldBeEnumNode(colorType, enumMember.Name);
+
+            Assert.True(leftNode.TypeReference.IsEnum());
+            Assert.Equal(enumMember.Name, leftNode.Value.ToString());
+            Assert.Equal(expectedLiteral, leftNode.LiteralText);
+
+            CollectionPropertyAccessNode rightNode = Assert.IsType<CollectionPropertyAccessNode>(inNode.Right);
+            rightNode.ShouldBeCollectionPropertyAccessQueryNode(this.GetIEdmProperty("Colors"));
+            Assert.Equal(colorType, rightNode.ItemType.Definition);
+        }
+
+        [Theory]
+        [InlineData("3.0")]
+        [InlineData("3.1")]
+        [InlineData("4.0")]
+        [InlineData("5.0")]
+        public void ParseFilterWithInOperatorWithEnumsMemberFloatIntegralValue_ThrowCollectionItemTypeMustBeSameAsSingleItemTypeException(string floatString)
+        {
+            // Arrange
+            string filterQuery = $"{floatString} in Colors";
+
+            // Act
+            Action test = () => ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+
+            // Assert
+            test.Throws<ArgumentException>(Strings.Nodes_InNode_CollectionItemTypeMustBeSameAsSingleItemType("NS.Color", "Edm.Single")); // Float are of Type Edm.Single
+        }
+
+        [Theory]
+        [InlineData(-20)]
+        [InlineData("-20")]
+        [InlineData("'-20'")]
+        public void ParseFilterWithInOperatorWithEnumsInvalidIntegralValues_ThrowsIsNotValidEnumConstantException(object integralValue)
+        {
+            // Arrange
+            string filterQuery = $"{integralValue} in Colors";
+
+            // Act
+            Action action = () => ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+
+            // Assert
+            action.Throws<ODataException>(Strings.Binder_IsNotValidEnumConstant("-20"));
+        }
+
+        [Fact]
+        public void ParseFilterWithInOperatorWithEnumsMemberNameWithoutSingleQuotes_ThrowsNullReferenceException()
+        {
+            // Arrange
+            string filterQuery = "Red in Colors";
+
+            string erroMessage = "Object reference not set to an instance of an object.";
+
+            // Act
+            Action action = () => ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+
+            // Assert
+            action.Throws<NullReferenceException>(erroMessage);

```

Comment:
Is the cause of the exception the fact that `Red` in the `filterQuery` is not in single quotes? If so, why does it throw a null reference exception? Is that by design or a bug? Should it throw a friendlier error message that states what the error is? (e.g. `Red` is not a known property or is not of the correct type or whatever the appropriate error should be)?

Reply:
@habbes Thanks for bringing this up. This was happening because the `EdmEntityType` for testing was set to be open. In case where isOpen is false, the exception is thrown: "`Could not find a property named '{1}' on type '{0}'.`"

```cs
internal SingleValueOpenPropertyAccessNode GeneratePropertyAccessQueryForOpenType(
EndPathToken endPathToken, SingleValueNode parentNode)
{
    if (parentNode.TypeReference == null ||
        parentNode.TypeReference.Definition.IsOpen() ||
        IsAggregatedProperty(endPathToken))
    {
        return new SingleValueOpenPropertyAccessNode(parentNode, endPathToken.Identifier);
    }
    else
    {
        throw new ODataException(ODataErrorStrings.MetadataBinder_PropertyNotDeclared(
            parentNode.TypeReference.FullName(),
            endPathToken.Identifier));
    }
}
```

I have modified the `isOpen` to `false`
```cs
this.entityType = new EdmEntityType("NS", "MyEntityType", isAbstract: false, isOpen: false, baseType: null);
```

Reply:
Okay that makes sense. But it seems like there are two issues. This has not addressed the issue when we have an open type, which I assume would still throw a null-reference exception. I'd still be curious about that, because it's possible that customers are using open types. I guess the challenge is that we may not know the type. But regardless, if it's an error, it should throw a meaningful error, not a null-reference exception.

Reply:
@habbes That seems like an existing issue. Even if we remove this fix, we will still get the same error. If you pass `Red` without single quotes, it will be treated like a property and hence the error when trying to check if `left node` can be assignableFrom `right node`. This is happening because the left node does not exist as a property and therefore has a `null` `EdmTypeReference`:

This image shows where the error is thrown:

![image](https://github.com/user-attachments/assets/6fb30785-8bf5-4fec-9bfd-a96ef8e27ba5)

I found that this is where the issue is in the case of `In Operator`: 
```js
if (!this.left.GetEdmTypeReference().IsAssignableFrom(this.right.ItemType) &&
!this.right.ItemType.IsAssignableFrom(this.left.GetEdmTypeReference()))
```
In the above code, `this.left.GetEdmTypeReference()` is always null for this example: `?$filter=Red in Colors`.

```cs
public InNode(SingleValueNode left, CollectionNode right)
{
    ExceptionUtils.CheckArgumentNotNull(left, "left");
    ExceptionUtils.CheckArgumentNotNull(right, "right");
    this.left = left;
    this.right = right;

    if (!right.GetEdmTypeReference().IsUntyped())
    {
        if (!this.left.GetEdmTypeReference().IsAssignableFrom(this.right.ItemType) &&
            !this.right.ItemType.IsAssignableFrom(this.left.GetEdmTypeReference()))
        {
            throw new ArgumentException(ODataErrorStrings.Nodes_InNode_CollectionItemTypeMustBeSameAsSingleItemType(
                this.right.ItemType.FullName(), this.left.GetEdmTypeReference().FullName()));
        }
    }
}
```

Reply:
@habbes Unless the spec permit, we can try to convert the left operand `Red` to the same enum type if possible. Hence, to support the following. But this will only work if the `EdmEntityType` is `open`
```js
?$filter=Red in Colors
```

Reply:
Since this PR was specific about the literals, may no need to add support for type coercions of non-literals. I was just concerned about the error message.